### PR TITLE
Display sub-items in course sidebar

### PR DESCRIPTION
**Pull request by korgan00**
## Changes

Fix https://github.com/Qiskit/saiba/issues/170

## Implementation details

Since the `walkTokens.ts` is growing. I created a new `walk-tokens` for moving all the `walkXXXX` functions in separate files.

The `RendererMarkdown` file is not used anymore in `UnitContent` since it only uses a function. Now, the function returns additional information for the application that doesn't make sense to have in it.

### CODE REVIEW

**Comment by abdonrd** [comment]
The jump is appreciated while the titles are not finished extracting:

https://user-images.githubusercontent.com/1007051/201761183-a321255d-870d-48bc-8b06-dc99a76c2ee7.mov

Maybe we should start moving this kind of logic to the server?

**Review by abdonrd — APPROVED** [review]
I'm not sure about the `walkHeadings` logic, but the other looks good!

**Comment by abdonrd** [comment]
Oh! Two things I just realized aren't working as expected:

- The links doesn't work; when I click in a sub-item, doesn't scroll me to the corresponding section.
- The active section is not highlighted.

**Review by abdonrd — CHANGES_REQUESTED** [review]
Oh! Two things I just realized aren't working as expected:

- The links doesn't work; when I click in a sub-item, doesn't scroll me to the corresponding section.
- The active section is not highlighted.

File: assets/scss/main.scss [hunk 1]
```diff
@@ -23,6 +23,7 @@
 html,
 body {
   overflow-x: hidden;
+  overflow-x: clip;
```
Comment (yoshiemuranaka) [inline assets/scss/main.scss:26]: isn't this declaration overriding the previous line?

**Review by yoshiemuranaka — COMMENTED** [review]
tested locally and works great! just left one comment

File: assets/scss/main.scss (hunk 1 above)
Comment (korgan00) [reply assets/scss/main.scss:26]: `clip` "only" has 81% of global usage, I am using `hidden` as fallback.

**Comment by korgan00** [comment]
> Oh! Two things I just realized aren't working as expected:
> 
> * The links doesn't work; when I click in a sub-item, doesn't scroll me to the corresponding section.
> * The active section is not highlighted.

Created a new PR to solve some of these comments.
If you agree, we can merge this PR.

File: assets/scss/main.scss (hunk 1 above)
Comment (yoshiemuranaka) [reply assets/scss/main.scss:26]: makes sense! maybe a comment would be helpful so it isn't accidentally refactored in the future.

```
  overflow-x: hidden; // fallback
```

File: assets/scss/main.scss [hunk 2]
```diff
@@ -23,6 +23,7 @@
 html,
 body {
   overflow-x: hidden;
```
Comment (korgan00) [inline assets/scss/main.scss:25]:
```suggestion
  overflow-x: hidden; // fallback
```

File: assets/scss/main.scss (hunk 1 above)
Comment (korgan00) [reply assets/scss/main.scss:26]: https://github.com/Qiskit/saiba/pull/292#discussion_r1025326825

**Review by yoshiemuranaka — APPROVED** [review]

**Comment by abdonrd** [comment]
Looks like it's not working?

https://saiba-pr-292.tfb1j0wd7x2.us-south.codeengine.appdomain.cloud/course/course-aa/unit-03

Screenshot 2022-11-17 at 2 31 23 PM

**Comment by abdonrd** [comment]
Ok, it's because the h1 it's now ignored!

**Review by abdonrd — APPROVED** [review]

**Pull request merged**